build_tools/cmake/iree_bitcode_library.cmake

```diff
@@ -78,13 +78,13 @@ function(iree_bitcode_library)
       OUTPUT
         "${_BITCODE_FILE}"
       COMMAND
-        "${IREE_CLANG_BINARY}"
+        "clang"
```

Comment:
There are other places using these CMake variables https://github.com/openxla/iree/blob/a623a93d27cd75b7c306a8282704f10b5a58ad3a/build_tools/cmake/iree_llvm.cmake#L74-L76 https://github.com/openxla/iree/blob/a623a93d27cd75b7c306a8282704f10b5a58ad3a/samples/custom_dispatch/cpu/embedded/CMakeLists.txt#L34-L39

I don't remember the full context here... but we should avoid having two different mechanisms for accessing the same tool

Reply:
The change in this PR uses the variables set in `iree_llvm.cmake`. I can make the changes in samples for the consistency.

The goal is of this PR to make ukernel bitcode gen working in the cross-compilation scenarios, so `clang` and `llvm-link` needs to be in `install/bin` and `iree_bitcode_library` macro can use binaries from IREE_HOST_BIN_DIR` instead from the CMake target list.